gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -0,0 +1,244 @@
+
+  def _add_annotation(self, field_name, data, header_desc):
+    """Adds an annotation INFO field based on the format in the header.
+
+    Args:
+      field_name (str): The name of the annotation field.
+      data(List[str]): The data part of the field separated on comma.
+      header_desc (str): The description of the field in the header used for
+        extracting the name of different annotations in `data`.
+    """
+    annotation_names = extract_annotation_names(header_desc)
+    for annotation_str in data:
+      annotations = extract_annotation_list_with_alt(annotation_str)
+      if len(annotation_names) != len(annotations) - 1:
+        raise ValueError('Expected {} annotations, got {}'.format(
+            len(annotation_names), len(annotations) - 1))
+      found_alt = False
+      # This assumes that number of alternate bases and annotation segments
+      # are not too big. If this assumption is not true, we should replace the
+      # following loop with a hash table search and avoid the quadratic time.
+      for alt in self._single_alternate_datas:
+        if alt.get_alternate_bases() == annotations[0]:
+          found_alt = True
+          alt._add_annotations(field_name, annotations[1:], annotation_names)
+          break
+      # TODO(bashir2): Currently we only check exact matches of alternate bases
+      # which is not enough. We should implement the whole standard for finding
+      # alternate bases for an annotation list.
+      if not found_alt:
+        logging.warning('Could not find matching alternate bases for {} in '
+                        'annotation filed {} with content {}'. format(
+            annotations[0], field_name, annotation_str))
+
+
+class SingleAlternateData(object):
+
+  def __init__(self, alt_bases):
+    self._alt_bases = alt_bases
+    self._infos = {}
+    # Each list item in this '_annotations' dictionary is dictionary itself.
+    self._annotations = defaultdict(lambda: [])
+
+  def get_alternate_bases(self):
+    return self._alt_bases
+
+  def get_infos(self):
+    return self._infos
+
+  def get_annotation_field_names(self):
+    return self._annotations.keys()
+
+  def get_all_annotations_for_field(self, field_name):
+    """
+
+    Returns:
+      `List[dict]`: Each member of the list is a map from annotation names
+        annotation values.
+    """
+    if not self._annotations.has_key(field_name):
+      raise ValueError('Annotation field {} not found.'.format(field_name))
+    return self._annotations[field_name]
+
+  def _add_single_info(self, field_name, data):
+    self._infos[field_name] = data
+
+  def _add_annotations(self, field_name, annotations, annotation_names):
+    annotation_dict = {}
+    for i in range(len(annotations)):
+      annotation_dict[annotation_names[i]] = annotations[i]
+    self._annotations[field_name].append(annotation_dict)
+
+
+def create_processed_variant(variant, header_fields,
```

Comment:
Consider writing this as a PTransform where input is a Variant and output is a ProcessedVariant. The ProcessedVariant class itself should just be a data storage class without any data manipulation logic (similar to the Variant class; think of it as a proto) and all such logic should be inside that PTransform or a helper library class ([recommended by Beam](https://beam.apache.org/contribute/ptransform-style-guide/#exposing-a-ptransform-vs-something-else)) I think it would be even cleaner to add a 'modify_schema' method to that library so that we can remove annotations and "split alt info logic" from bigquery_vcf_schema altogether (this would be similar to variant_merger). That way, if we decide to add a new annotation field in the future, all of the logic would be in one place.